### PR TITLE
Updates material-ui and react npm libraries

### PR DESCRIPTION
**Pull request by rileyjbauer**
Most changes are adding `variant='outlined'` to all of the inputs. It seems we can no longer extend `TextFieldProps` directly due to its lack of compile time typing: "An interface can only extend an object type or intersection of object types with statically known members."
We can extend `OutlinedTextFieldProps` directly to avoid this, but it has `variant` as a required field, hence all of the additions.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/630)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -12,10 +12,11 @@
     "dagre": "^0.8.2",
     "http-proxy-middleware": "^0.19.0",
     "immer": "^1.7.4",
+    "jss": "^10.0.0-alpha.3",
```
Comment (yebrahim) [inline frontend/package.json]: Are we using jss?

Comment (rileyjbauer) [reply frontend/package.json]: No, this is a workaround for https://github.com/mui-org/material-ui/issues/14040. Once that issue is fixed, we should be able to remove this direct dep

Comment (yebrahim) [reply frontend/package.json]: Ah interesting.
You can consider using an older version (e.g. 3.7.x) that moves us very close to being updated without needing this hack. Keep in mind that even with the latest, we'll be outdated again in just a week or two. Up to you tho.

Comment (rileyjbauer) [reply frontend/package.json]: I set it to 3.7.1 so we don't need the workaround, but we'll need to remember to update it in the future

**Pull request merged**